test/SNLS_testdriver_hybrd.cc

```diff
@@ -0,0 +1,368 @@
+         temp1 = 1.0;
+         temp2 = 2.0 * x[j] - 1.0;
+         temp = 2.0 * temp2;
+         for (int i = 0; i < n; i++) {
+               r[i] += temp2;
+               ti = temp * temp2 - temp1;
+               temp1 = temp2;
+               temp2 = ti;
+         }
+      }
+
+      tk = 1.0 / ((double) n);
+      iev = -1;
+      for (int k = 0; k < n; k++) {
+         r[k] = tk * r[k];
+         if (iev > 0) {
+               d1 = ((double) k) + 1.0;
+               r[k] += 1.0 / (d1 * d1 - 1.0);
+         }
+         iev = -iev;
+      }
+      if (J != nullptr) {
+         tk = 1.0 / ((double) n);
+         for (int j = 0; j < n; j++) {
+            temp1 = 1.0;
+            temp2 = 2.0 * x[j] - 1.0;
+            temp = 2.0 * temp2;
+            temp3 = 0.0;
+            temp4 = 2.0;
+            for (int k = 0; k < n; k++) {
+                  J[SNLS_NN_INDX(j, k, nDimSys)] = tk * temp4;
+                  ti = 4. * temp2 + temp * temp4 - temp3;
+                  temp3 = temp4;
+                  temp4 = ti;
+                  ti = temp * temp2 - temp1;
+                  temp1 = temp2;
+                  temp2 = ti;
+            }
+         }
+      }
+            
+    return true;
+   }
+
+};
+
+// This test will fail due to our delta being too aggressive which causes the solution to not move 
+TEST(snls, chebyquad_a) // int main(int , char ** )
+{
+   const int nDim = ChebyQuad<3>::nDimSys;
+
+   ChebyQuad<3> chebyquad; // LAMBDA_BROYDEN 
+   snls::SNLSHybrdTrDLDenseG<ChebyQuad<3>> solver(chebyquad);
+   snls::TrDeltaControl deltaControl;
+   deltaControl._deltaInit = 1e0;
+   deltaControl._xiDecDelta = 0.75;
+   solver.setupSolver(NL_MAXITER, NL_TOLER, &deltaControl, 1);
+  
+   double h = 1.0 / (((double) nDim) + 1.0);
+   for (int j = 0; j < nDim; j++){
+      solver.m_x[j] = (((double) j) + 1.0) * h;
+   }
+   //
+   double r[nDim], J[nDim*nDim] ;
+   //
+   // any of these should be equivalent:
+   // broyden.computeRJ(r, J, solver._x);
+   // solver._crj.computeRJ(r, J, solver._x);
+   solver.computeRJ(r, J); 
+
+   snls::SNLSStatus_t status = solver.solve( );
+   std::cout << "Status " << status << std::endl;
+   EXPECT_FALSE( snls::isConverged(status) ) << "Expected solver to fail" ;
+   if ( status != snls::converged ){
+      char errmsg[256];
+      snprintf(errmsg, sizeof(errmsg), "Solver failed to converge! Using tol=%g and maxIter=%i",NL_TOLER,NL_MAXITER);
+      SNLS_WARN(__func__,errmsg);
+   }
+/*
+   std::cout << "Function evaluations: " << solver.getNFEvals() << "\n";
+   std::cout << "Jacobian evaluations: " << solver.getNJEvals() << "\n";
+   EXPECT_EQ( solver.getNFEvals(), 30) << "Expected 29 function evaluations for this case";
+   EXPECT_EQ( solver.getNJEvals(), 3) << "Expected 3 jacobian evaluations for this case";
+*/
+}
+
+TEST(snls, chebyquad_b) // int main(int , char ** )
+{
+   const int nDim = ChebyQuad<5>::nDimSys;
+
+   ChebyQuad<5> chebyquad; // LAMBDA_BROYDEN 
```

Comment:
Here and elsewhere, what is the point of the comment with `LAMBDA_BROYDEN` by the `chebyquad` declaration?

Reply:
This was a copy and paste of test suites. I'll update this and remove that comment.